kie-server-parent/kie-server-services/kie-server-services-kafka/src/test/java/org/kie/server/services/jbpm/kafka/KakfaServerExtensionTest.java

```diff
@@ -0,0 +1,149 @@
+
+        protected Consumer<String, byte[]> getKafkaConsumer() {
+            return consumer;
+        }
+
+        public MockConsumer<String, byte[]> getMockConsumer() {
+            return consumer;
+        }
+    }
+
+    private ProcessService processService;
+    private MockKafkaServerExtension extension;
+    private KieServerImpl server;
+    private KieServerRegistry registry;
+    private DeployedUnit deployedUnit;
+    private ProcessDefinition processDefinition;
+
+    @Before
+    public void setup() {
+        extension = new MockKafkaServerExtension();
+        server = mock(KieServerImpl.class);
+        registry = mock(KieServerRegistry.class);
+        KieServerExtension serverExtension = mock(KieServerExtension.class);
+
+        when(registry.getServerExtension(Mockito.anyString())).thenReturn(serverExtension);
+        ListenerSupport deployService = mock(ListenerSupport.class, withSettings().extraInterfaces(
+                DeploymentService.class));
+        processService = mock(ProcessService.class);
+        when(serverExtension.getServices()).thenReturn(Arrays.asList(deployService, processService));
+        extension.init(server, registry);
+        extension.serverStarted();
+        deployedUnit = mock(DeployedUnit.class);
+        processDefinition = mock(ProcessDefinition.class);
+        when(deployedUnit.getDeployedAssets()).thenReturn(Collections.singletonList(processDefinition));
+    }
+
+    @After
+    public void close() {
+        extension.destroy(server, registry);
+    }
+
+    @Test
+    public void testKafkaServerExecutorSignal() throws InterruptedException {
+        when(processDefinition.getSignalsMetadata()).thenReturn(Collections.singletonList(SignalDescImpl.from(
+                new Signal("MySignal", "MySignal", "String"))));
+        when(deployedUnit.getDeployedAssets()).thenReturn(Collections.singletonList(processDefinition));
+        extension.onDeploy(new DeploymentEvent("MyDeploy", deployedUnit));
+        publishEvent("MySignal", "{\"id\":\"javi\",\"type\":\"one\",\"source\":\"pepe\",\"data\":\"javierito\"}");
+        verify(processService, getTimeout()).signalEvent("MyDeploy", "MySignal", "javierito");
+    }
+
+    @Test
+    public void testKafkaServerExecutorMessage() throws InterruptedException {
+
+        Message msg = new Message("MyMessage");
+        msg.setName("Hello");
+        msg.setType("String");
+        when(processDefinition.getSignalsMetadata()).thenReturn(Collections.singletonList(SignalDescImpl.from(msg)));
+        extension.onDeploy(new DeploymentEvent("MyDeploy", deployedUnit));
+        publishEvent("Hello", "{\"id\":\"javi\",\"type\":\"one\",\"source\":\"pepe\",\"data\":\"pepe\"}");
+        verify(processService, getTimeout()).signalEvent("MyDeploy", "Message-Hello", "pepe");
+    }
+
+    @Test
+    public void testKafkaServerExecutorMessageTopic() throws InterruptedException {
```

Comment:
Checked exception is not thrown within this method

```suggestion
    public void testKafkaServerExecutorMessageTopic() {
```

Reply:
good catch